### ISSUE TITLE
Make `quoting.js` less destructive

### ISSUE THREAD

**Issue by siefkenj** [protz/thunderbird-conversations#1435]
`quoting.js` seems to modify the DOM of the message iframe directly, which makes it hard to test and revert if an error is thrown. I'm no sure how much DOM manipulation conversations does, but I would propose a function that inputs a document fragment and returns a new document fragment with the blockquote and then replacing the body of the iframe instead of modifying it directly. This could also lead to a "turn off smart formatting" option.

There are also various comments around `quoting.js` that say it almost never works. Are these heuristics worth including? Do newer Yahoo and Hotmail messages have more predictable quoting?

**Comment by Standard8** [comment]
@siefkenj I think you've done this now, or is there more to do?

**Comment by siefkenj** [comment]
Yes and no...the capability is there, but its still being used in a destructive way.
